Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs

```diff
@@ -66,35 +66,53 @@ public class DataStreamFacade : IDataStreamFacade
         /// <returns>the streams of user data.</returns>
         public async IAsyncEnumerable<IEnumerable<UserData>> GetUserDataStreamAsync(string notificationId)
         {
-            if (notificationId == null)
-            {
-                throw new ArgumentNullException(nameof(notificationId));
-            }
+            _ = notificationId ?? throw new ArgumentNullException(nameof(notificationId));
 
             var sentNotificationDataEntitiesStream = this.sentNotificationDataRepository.GetStreamsAsync(notificationId);
+            var isForbidden = false;
+
             await foreach (var sentNotifcations in sentNotificationDataEntitiesStream)
             {
-                List<User> userList = new List<User>();
+                var users = new List<User>();
+
+                // filter the recipient not found users.
+                var recipients = sentNotifcations.Where(sentNotifcation => !sentNotifcation.DeliveryStatus.Equals(SentNotificationDataEntity.RecipientNotFound, StringComparison.CurrentCultureIgnoreCase));
```

Comment:
Is DeliveryStatus localized? Or should the comparison here be InvariantCultureIgnoreCase?

Reply:
DeliveryStatus is not localized when saved in DB. It is localized only at the time it is exported.